datafusion/sqllogictest/test_files/operator.slt

```diff
@@ -52,6 +52,42 @@ from numeric_types;
 ----
 Int8 Int16 Int32 Int64 UInt8 UInt16 UInt32 UInt64 Float32 Float64 Decimal128(6, 2)
 
+# Plus with literal integer
+query TTTTTTTTTTT
+select
+    arrow_typeof(int8 + 2),
+    arrow_typeof(int16 + 2),
+    arrow_typeof(int32 + 2),
+    arrow_typeof(int64 + 2),
+    arrow_typeof(uint8 + 2),
+    arrow_typeof(uint16 + 2),
+    arrow_typeof(uint32 + 2),
+    arrow_typeof(uint64 + 2),
+    arrow_typeof(float32 + 2),
+    arrow_typeof(float64 + 2),
+    arrow_typeof(decimal + 2)
+from numeric_types;
+----
+Int64 Int64 Int64 Int64 Int64 Int64 Int64 Int64 Float32 Float64 Decimal128(23, 2)
```

Comment:
This shows that adding a literal integer (`2`) does not upcast to Decimal for integer types (which would be terrible for performance)